integration/aggregate_documents_compat_test.go

```diff
@@ -1562,3 +1562,47 @@ func TestAggregateCompatProject(t *testing.T) {
 
 	testAggregateStagesCompat(t, testCases)
 }
+
+func TestAggregateCompatUnset(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]aggregateStagesCompatTestCase{
+		"InvalidType": {
+			pipeline: bson.A{
+				bson.D{{"$unset", "invalid"}},
+			},
+			resultType: emptyResult,
+		},
+		"Unset1Field": {
+			pipeline: bson.A{
+				bson.D{{"$unset", "v"}},
+			},
+		},
+		"UnsetID": {
+			pipeline: bson.A{
+				bson.D{{"$unset", "_id"}},
+			},
+		},
+		"Unset2Fields": {
```

Comment:
I suggest also adding a test case where we unset both `_id` and `v` fields. So, in the results, we will see some empty docs for some providers.